### DISCUSSION TITLE
RTSP Stream Control Plugin

### DISCUSSION THREAD

**Discussion by pedrosbanioles** [ant-media/Ant-Media-Server#6356] [Plugins/Custom Extensions]
According to RFC 2326 some RTSP Servers can support modification of the scale through the header during a stream to facilitate changing the playback speed. I have tried this using a dotnet application that uses [SharpRTSP](https://github.com/ngraziano/SharpRTSP) and it works when connecting to Hikvision DVRs.

My Question would be, what is the feasibility of setting up a plugin to offer the ability to send requests during stream to the rtsp server to do this, or to add a Java base RTSP client that could be hooked into to provide this facility.

I can see that the StreamFetcher is using FFMpeg to do the connection, but is that a closed book with regard to sending further rtsp PLAY requests during a stream? What would be the performance impact of implementing a Java based rtsp client to pull and also control the rtsp session?

**Comment by yashtandon113** [comment]
@pedrosbanioles 
As discussed over the ticket, we will discuss this question in our meeting.

**Reply by mekya** [reply]
Hi @yashtandon113,

Could you please give an update here as well?

Cheers
Oz
